libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/performance/KotlinPerformancePlugin.kt

```diff
@@ -66,17 +66,26 @@ open class KotlinPerformancePlugin : Plugin<Project> {
 
     private fun configureTasks(project: Project, performanceExtension: PerformanceExtension) {
         // Add time listener.
-        val timeListener = TaskTimerListener(project)
-        project.gradle.addListener(timeListener)
-        performanceExtension.trackedBinaries.forEach { binary ->
-            val perfReport = project.registerTask<NativePerformanceReport>(binary.target.disambiguateName(lowerCamelCaseName("perfReport", binary.name))) {
-                it.binary = binary
-                it.settings = performanceExtension
-                it.timeListener = timeListener
-                it.group = TASK_GROUP
-                it.description = "Report performance measurement results for binary '${binary.name}' of target '${binary.target.name}'."
+        if (!isConfigurationCacheAvailable(project.gradle)) {
+            val timeListener = TaskTimerListener(project)
+            project.gradle.addListener(timeListener)
```

Comment:
You can't use a `TaskExecutionListener` with the configuration cache, but you can replace it with a build service as a `OperationCompletionListener`, and it'll work both without and with the configuration cache.
See https://docs.gradle.org/release-nightly/userguide/build_services.html#operation_listener